src/blockdev.rs

```diff
@@ -1569,17 +1643,35 @@ mod tests {
 
         // test trying to save partitions from a MBR disk
         let mut disk = make_unformatted_disk();
-        gptman::GPT::write_protective_mbr_into(&mut disk, 512).unwrap();
-        // label only
-        SavedPartitions::new(&mut disk, 512, &vec![label("*i*")]).unwrap();
+        let mut mbr = mbrman::MBR::new_from(&mut disk, 512 as u32, [0xff; 4])
+            .expect("could not create partition table");
+
+        // create a mbr partition to copy over to gpt
+        mbr[1] = mbrman::MBRPartitionEntry {
+            boot: mbrman::BOOT_INACTIVE,
+            first_chs: mbrman::CHS::empty(),
+            sys: 0x0f,
+            last_chs: mbrman::CHS::empty(),
+            starting_lba: 1,
+            sectors: mbr.disk_size - 5000,
+        };
+
+        mbr.write_into(&mut disk).unwrap();
         // index only
-        assert_eq!(
-            SavedPartitions::new(&mut disk, 512, &vec![Index(index(1), index(1))])
-                .unwrap_err()
-                .to_string(),
-            "saving partitions from an MBR disk is not yet supported"
-        );
-        // label and index
+        let saved = SavedPartitions::new(&mut disk, 512, &vec![Index(index(1), index(1))]).unwrap();
+        assert!(saved.is_saved());
```

Comment:
We're only checking that we think we succeeded, but we're not actually verifying the results against expected values.  Let's fix that.  In particular, we should check the resulting type GUIDs.

We should be testing a non-trivial MBR layout, where we aren't asking to save all of the partitions.  That layout should include logical partitions.  And we should also test saving a wildcard label `*` (which we might decide should match either everything or nothing).

We should also test with a 4Kn disk.